PhpAmqpLib/Wire/IO/AbstractIO.php

```diff
@@ -82,11 +82,22 @@ public function select($sec, $usec)
     {
         $this->check_heartbeat();
         $this->set_error_handler();
-        try {
-            $result = $this->do_select($sec, $usec);
-            $this->cleanup_error_handler();
-        } catch (\ErrorException $e) {
-            throw new AMQPIOWaitException($e->getMessage(), $e->getCode(), $e);
+
+        if (PHP_VERSION_ID >= 80000) {
+            try {
+                $result = $this->do_select($sec, $usec);
+                $this->cleanup_error_handler();
+            } catch (\Throwable $e) {
```

Comment:
Why do we need to have separate try catch blocks? BTW, we are dropping PHP5.6 suuport, so let's one \Twroable block with additional check.

Reply:
It is not for php 5.6. It is for php 8:
php 8 throws [ValueError](https://www.php.net/manual/en/class.valueerror.php) in this particular place. 
So correct `catch` block would be `catch (\ErrorException|\ValueError $ex)` (as it is said in the comments).
But such syntax is not supported by php 7.0 which is still supported. When 7.1 become minimum supported version this code could be removed.

Reply:
https://www.php.net/supported-versions.php

PHP v7.3 is the minimal supported by the php community version. Is it necessary to support versions of PHP that would never get security updates?

Reply:
Why it's ValueError, where does this exception gets thrown? Are You sure we can replace ValueError with AMQPIOWaitException?

Reply:
@masterjus because there is still many with old PHP versions in production, despite it's not supported any more. Not everyone can upgrade easily.

Reply:
@ramunasd I have reverted this commit => you can see the failure: https://github.com/php-amqplib/php-amqplib/pull/858/checks?check_run_id=1904199648

Reply:
Thanks @axxapy, this is a new issue, specific to PHP8. I will resolve it seperatelly as #883

Reply:
So, what should I do?
Let me remove my revert so tests would pass => You'll fix it in a separate PR properly?

Reply:
Please keep it as it was orginally, thanks.

Reply:
kk, I have rebased, dropped my commit. 
reference, just for the case: db1f905